packages/compartment-mapper/src/link.js

```diff
@@ -372,6 +397,10 @@ export const link = (
     // The `moduleMapHook` writes back to the compartment map.
     compartmentDescriptor.modules = modules;
 
+    for (const [extension, language] of entries(customLanguageForExtension)) {
+      languageForExtension[extension] = language;
+    }
```

Comment:
The `parsers` must take precedence over or override the default language for extension. We should have a test to confirm this.